spring-cloud-azure-config/src/main/java/com/microsoft/azure/spring/cloud/config/ConfigServiceTemplate.java

```diff
@@ -0,0 +1,59 @@
+/*
+ * Copyright (c) Microsoft Corporation. All rights reserved.
+ * Licensed under the MIT License. See LICENSE in the project root for
+ * license information.
+ */
+package com.microsoft.azure.spring.cloud.config;
+
+import java.io.IOException;
+import java.net.URISyntaxException;
+import java.util.List;
+
+import com.fasterxml.jackson.databind.ObjectMapper;
+import com.fasterxml.jackson.databind.node.ObjectNode;
+import com.microsoft.azure.spring.cloud.config.domain.KeyValueItem;
+import lombok.AllArgsConstructor;
+import lombok.extern.slf4j.Slf4j;
+import org.apache.http.HttpStatus;
+import org.apache.http.client.methods.CloseableHttpResponse;
+import org.apache.http.client.methods.HttpGet;
+
+import org.springframework.lang.Nullable;
+
+@Slf4j
+@AllArgsConstructor
+public class ConfigServiceTemplate implements ConfigServiceOperations {
+    private static final ObjectMapper mapper = new ObjectMapper();
+
+    private final ConfigHttpClient configClient;
+    private final String storeEndpoint;
+    private final String credential;
+    private final String secret;
+
+    @Override
+    public List<KeyValueItem> getKeys(@Nullable String prefix, @Nullable String label) {
+        String requestUri = new RestAPIBuilder().withEndpoint(storeEndpoint).buildKVApi(prefix, label);
+        HttpGet httpGet = new HttpGet(requestUri);
```

Comment:
Shall we add a `debug` log here printing the final URL?

Reply:
Added the debug log and updated the exception message.